ingest/src/main/java/com/connexta/ingest/client/StoreClient.java

```diff
@@ -43,13 +44,26 @@ public StoreClient(
   public URI store(
       @NotNull @Min(1L) @Max(10737418240L) final Long fileSize,
       @NotEmpty final String mimeType,
-      @NotNull final MultipartFile file,
+      @NotNull final InputStream inputStream,
       @NotEmpty final String fileName)
       throws StoreException {
     final MultipartBodyBuilder builder = new MultipartBodyBuilder();
     builder.part("fileSize", fileSize);
     builder.part("mimeType", mimeType);
-    builder.part("file", file.getResource());
+    builder.part(
+        "file",
+        new InputStreamResource(inputStream) {
```

Comment:
I wish there was a nicer way to do this

Reply:
I agree. I didn't research it much, so maybe I missed something.